internal/rpc/rpcserver/server.go

```diff
@@ -3322,6 +3322,132 @@ func (s *votingServer) SetVoteChoices(ctx context.Context, req *pb.SetVoteChoice
 	return resp, nil
 }
 
+// TSpendPolicies returns voting policies for particular treasury spends
+// transactions. If a tspend transaction hash is specified, that policy is
+// returned; otherwise the policies for all known tspends are returned in an
+// array.  If both a tspend transaction hash and a ticket hash are provided,
+// the per-ticket tspend policy is returned.
+func (s *votingServer) TSpendPolicies(ctx context.Context, req *pb.TSpendPoliciesRequest) (*pb.TSpendPoliciesResponse, error) {
+	var ticketHash *chainhash.Hash
+	var hash *chainhash.Hash
+	var err error
+	if len(req.TicketHash) != 0 {
+		ticketHash, err = chainhash.NewHash(req.TicketHash)
+		if err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, "%v", err)
+		}
+	}
+
+	if len(req.Hash) != 0 {
+		hash, err = chainhash.NewHash(req.Hash)
+		if err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, "%v", err)
+		}
+		var policy string
+		switch s.wallet.TSpendPolicy(hash, ticketHash) {
+		case stake.TreasuryVoteYes:
+			policy = "yes"
+		case stake.TreasuryVoteNo:
+			policy = "no"
+		default:
+			policy = "abstain"
+		}
+		resp := &pb.TSpendPoliciesResponse{
+			Policies: make([]*pb.TSpendPoliciesResponse_Policy, 0, 1),
+		}
+		r := &pb.TSpendPoliciesResponse_Policy{
+			Hash:       req.Hash,
+			Policy:     policy,
+			TicketHash: []byte{},
+		}
+		if req.TicketHash != nil {
+			r.TicketHash = req.TicketHash
+		}
+		resp.Policies = append(resp.Policies, r)
+		return resp, nil
+	}
+
+	tspends := s.wallet.GetAllTSpends(ctx)
+	resp := &pb.TSpendPoliciesResponse{
+		Policies: make([]*pb.TSpendPoliciesResponse_Policy, 0, len(tspends)),
+	}
+	for i := range tspends {
+		tspendHash := tspends[i].TxHash()
+		p := s.wallet.TSpendPolicy(&tspendHash, ticketHash)
+
+		var policy string
+		switch p {
+		case stake.TreasuryVoteYes:
+			policy = "yes"
+		case stake.TreasuryVoteNo:
+			policy = "no"
+		}
+		r := &pb.TSpendPoliciesResponse_Policy{
+			Hash:       tspendHash[:],
+			Policy:     policy,
+			TicketHash: []byte{},
+		}
+		if req.TicketHash != nil {
+			r.TicketHash = req.TicketHash
+		}
+		resp.Policies = append(resp.Policies, r)
+	}
+	return resp, nil
+}
+
+// SetTSpendPolicy saves the voting policy for a particular tspend transaction
+// hash, and optionally, setting the tspend policy used by a specific ticket.
+//
+// If a VSP host is configured in the application settings, the voting
+// preferences will also be set with the VSP.
```

Comment:
This doesn't appear to be the case.  Besides, we will want to use the VSP that is associated with each live ticket to be updated, and not the one in the current application settings.

Reply:
looks like this is being handled by another rpc call, so this comment can just be nuked and then this looks ok to me.